app/task_routes.py

```
@@ -0,0 +1,122 @@
+
+# validate ids for models
+def validate_id(cls, model_id):
+    try: 
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message":f"{cls.__name__} {model_id} invalid"}, 400))
+    
+    model = cls.query.get(model_id)
+    if not model:
+        abort(make_response({"message":f"{cls.__name__} {model_id} not found"}, 404))
+
+    return model
+
+# validate request body information
+def validate_request_body(cls, request_body):
+    try:
+        new_model = cls.from_dict(request_body)
+    except:
+        abort(make_response({"details":f"Invalid data"}, 400))
+
+    return new_model
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+
+    new_task = validate_request_body(Task, request_body)
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return make_response(jsonify({"task": new_task.to_dict()}), 201)
+
+@tasks_bp.route("", methods=["GET"])
+def read_all_tasks():
+    sort_query = request.args.get("sort")
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc()).all()
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all()
+    else:
+        tasks = Task.query.all()
+
+    tasks_response = [task.to_dict() for task in tasks]
+
+    return jsonify(tasks_response)
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    task = validate_id(Task, task_id)
+    if not task.goal_id:
+        response = {"task": task.to_dict()}
+    else:
+        response = {"task": task.to_dict_with_goal()}
+    
+    return make_response(jsonify(response))
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_id(Task, task_id)
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    task.completed_at = request_body.get("completed_at", None)
```

Can we leverage your validate_request_body() helper here like you do in your POST method?

These keys are required and should return an error message if they are left off.